62_CAD/shaders/main_pipeline/fragment_shader.hlsl

```diff
@@ -1,655 +1,602 @@
+                bool isCellPartiallyInvalid = isnan(cellHeights.x) || isnan(cellHeights.y) || isnan(cellHeights.z) || isnan(cellHeights.w);
+
+                if (isTriangleInvalid)
+                    discard;
+
+                // move from grid space to screen space
+                [unroll]
+                for (int i = 0; i < 3; ++i)
+                    v[i].xy += topLeft;
+
+                if (triangleA)
+                {
+                    outlineLineSegments[0] = nbl::hlsl::shapes::Line<float>::construct(v[2].xy, v[0].xy);
+                    outlineLineSegments[1] = nbl::hlsl::shapes::Line<float>::construct(v[2].xy, v[1].xy);
+                }
+                else
+                {
+                    outlineLineSegments[0] = nbl::hlsl::shapes::Line<float>::construct(v[1].xy, v[2].xy);
+                    outlineLineSegments[1] = nbl::hlsl::shapes::Line<float>::construct(v[0].xy, v[2].xy);
+                }
+
+                // test diagonal draw
+                //outlineLineSegments[0] = nbl::hlsl::shapes::Line<float>::construct(v[0].xy, v[1].xy);
+                //outlineLineSegments[1] = nbl::hlsl::shapes::Line<float>::construct(v[0].xy, v[1].xy);
+
+
+                float distancesToVerticalCellSides = min(insideCellCoord.x, cellWidth - insideCellCoord.x);
+                float distancesToHorizontalCellSides = min(insideCellCoord.y, cellWidth - insideCellCoord.y);
+
+                float patternCellCoord = distancesToVerticalCellSides >= distancesToHorizontalCellSides ? cellCoords.x : cellCoords.y;
+
+                float reciprocalPatternLength = input.getGridDTMOutlineStipplePatternLengthReciprocal();
+                if(reciprocalPatternLength > 0.0f)
+                    outlinePhaseShift = (cellWidth * (1.0f / globals.screenToWorldRatio) * patternCellCoord) * reciprocalPatternLength;
+            }
+
+            const float3 baryCoord = dtm::calculateDTMTriangleBarycentrics(v[0], v[1], v[2], input.position.xy);
+            float height = baryCoord.x * v[0].z + baryCoord.y * v[1].z + baryCoord.z * v[2].z;
+            float2 heightDeriv = fwidth(height);
+
+            const bool outOfBoundsUV = uv.x < 0.0f || uv.y < 0.0f || uv.x > 1.0f || uv.y > 1.0f;
+            float4 dtmColor = float4(0.0f, 0.0f, 0.0f, 0.0f);
+            if (dtmSettings.drawContourEnabled())
+            {
+                for (int i = dtmSettings.contourSettingsCount-1u; i >= 0; --i) 
+                    dtmColor = dtm::blendUnder(dtmColor, dtm::calculateDTMContourColor(dtmSettings.contourSettings[i], v, input.position.xy, height));
+            }
+            if (dtmSettings.drawOutlineEnabled())
+                dtmColor = dtm::blendUnder(dtmColor, dtm::calculateGridDTMOutlineColor(dtmSettings.outlineLineStyleIdx, outlineLineSegments, input.position.xy, outlinePhaseShift));
+            if (dtmSettings.drawHeightShadingEnabled() && !outOfBoundsUV)

```

Comment:
Ok I can live with this. just know that you're implicitly discarding height shading for out of bounds UVs. if your grid was rotated you'd get aliasing.

Reply:
no, it needs to be fixed, it causes this:
![image](https://github.com/user-attachments/assets/317494dc-d2d9-4292-8ab1-06e5d721746a)